### DISCUSSION TITLE
Web interface auth timeout

### DISCUSSION THREAD

**Discussion by BeecherNetworks** [TriliumNext/Trilium#2720] [Q&A] [answered]
Is there a way of adjusting the web interface auth timeout? I'm the only user so I'd like to extend it way out.

**Comment by sigaloid** [comment, marked as answer]
You can edit the code here: https://github.com/zadam/trilium/blob/eba824a5b1ff2c2b33f75a9d2eb01aadae2e13d7/src/routes/routes.js#L196-L199 but it's not currently configurable with settings.

**Reply by BeecherNetworks** [reply]
Thanks. I'm running in Docker so I'll leave well alone for the moment. I've opened a feature request [here](https://github.com/zadam/trilium/issues/2722).

**Reply by zadam** [reply]
That's not what is asked though - this is about throttling the rate of incoming login requests (protection against brute force attacks).

This should be it:

https://github.com/zadam/trilium/blob/77ce56ba840300fd79b4573dbbee339175944096/src/app.js#L43-L46

**Reply by sigaloid** [reply]
loginRateLimiter is being applied to the /login route, though, so wouldn't that be the rate of incoming login requests?

**Reply by zadam** [reply]
Sure, but the OP isn't asing about rate limiting, but about session expiration (at least that's what I understand).

**Reply by sigaloid** [reply]
Oh, you're right, I think I misread it. Thanks!
